web/annotation_layer_builder.js

```diff
@@ -119,6 +119,20 @@ class AnnotationLayerBuilder {
       return;
     }
 
+    const uniqueLinks = linkAnnotations.filter(link => {
+      for (const annotation of annotations) {
+        if (
+          annotation.subtype === "Link" &&
+          annotation.url === link.url &&
+          Util.intersect(annotation.rect, link.rect) !== null
```

Comment:
In order to avoid some corner case bug, maybe a better way to do would be to compute the ratio area(intersection) / area(annotation.rect) and if it's greater than a threshold then consider that links are very likely the same.